### PR TITLE
[Task #50232] Login logout API

### PR DESCRIPTION
**Pull request by ghost**
## Related Tickets:
[#50232](https://edu-redmine.sun-asterisk.vn/issues/50232)

## WHAT this PR does?
- Login API
- Logout API

## Notes Impacted Areas:
- N/A.

## Attachments:
image
image

### CODE REVIEW

**Comment by ghost** [comment]
ready

**Review by ghost — APPROVED** [review]

File: app/Http/Controllers/Api/AuthController.php [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+class AuthController extends Controller
+{
+    protected $userRepo;
+
+    public function __construct(UserRepositoryInterface $userRepo)
+    {
+        $this->userRepo = $userRepo;
+    }
+
+    public function login(LoginRequest $request)
```
Comment (nampt-3149) [inline app/Http/Controllers/Api/AuthController.php:19]: Form Request Validate fail thì sao

File: app/Http/Controllers/Api/AuthController.php [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+            }
+
+            return response()->json([
+                'status_code' => 500,
+                'message' => __('login-fail'),
+            ]);
+        } catch (\Exception $exception) {
+            return response()->json([
+                'status_code' => 500,
+                'error' => $exception,
```
Comment (nampt-3149) [inline app/Http/Controllers/Api/AuthController.php]:
```suggestion
                'error' => $exception->getMessage(),
```

File: app/Http/Controllers/Api/AuthController.php [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+    }
+
+    public function login(LoginRequest $request)
+    {
+        try {
+            $user = $this->userRepo->findUserByEmail($request->email);
+
+            if (Auth::attempt([
+                'email' => $request->input('email'),
+                'password' => $request->input('password'),
```
Comment (nampt-3149) [inline app/Http/Controllers/Api/AuthController.php:31]: check thử attempt sai thì trả về redirect 30x hay json

**Comment by ghost** [comment]
ready

**Review by nampt-3149 — APPROVED** [review]

**Pull request merged**